server.js

```diff
@@ -98,10 +98,18 @@ var icons = ["bear", "ox", "flamingo", "panda", "giraffe", "raccoon", "chimpanze
              "brownbear", "marmoset", "funnylion", "deer", "zebra", "meerkat", "elephant", "cat",
              "hare", "puma", "owl", "antelope", "lion", "fox", "wolf", "hippo"];
 
-io.on('connection', (socket) => {  
+num_users = 0;
```

Comment:
Can we declare this with let? Also, currently this is never decremented and just gets incremented forever as long as there are no icons available. Is that the behavior we want? It might be, but it looks like you were thinking of decrementing it on line 150.

Reply:
renamed this counter to overflow_id in 3687888

we never decrement to prevent the same number appearing for two users connected at the same time. we could instead default to socket.id for a truly unique id that we don't have to manage but that's less readable to an admin trying to quickly distinguish between chats